Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/audit/transform/CoreX12AuditDataTransform.java

```diff
@@ -0,0 +1,467 @@
+                return true;
+            }
+        } else if (msg instanceof COREEnvelopeBatchSubmissionResponse) {
+            coreEnvelopeBatchSubmissionResponse = (COREEnvelopeBatchSubmissionResponse) msg;
+            if (NullChecker.isNullish(coreEnvelopeBatchSubmissionResponse.getCORERuleVersion())) {
+                LOG.error("CORE X12 Batch CORERuleVersion is empty...");
+                return true;
+                //TODO: For some cases the payload id coming as null. Need to fix it as part of
+                //Batch Request and Batch Response fix
+            //} else if (NullChecker.isNullish(coreEnvelopeBatchSubmissionResponse.getPayloadID())) {
+            //    LOG.error("CORE X12 Batch PayloadID is empty...");
+            //    return true;
+            } else if (NullChecker.isNullish(coreEnvelopeBatchSubmissionResponse.getPayloadType())) {
+                LOG.error("CORE X12 Batch PayloadType is empty...");
+                return true;
+            } else if (NullChecker.isNullish(coreEnvelopeBatchSubmissionResponse.getProcessingMode())) {
+                LOG.error("CORE X12 Batch ProcessingMode is empty...");
+                return true;
+            } else if (NullChecker.isNullish(coreEnvelopeBatchSubmissionResponse.getReceiverID())) {
+                LOG.error("CORE X12 Batch ReceiverID is empty...");
+                return true;
+            } else if (NullChecker.isNullish(coreEnvelopeBatchSubmissionResponse.getSenderID())) {
+                LOG.error("CORE X12 Batch SenderID is empty...");
+                return true;
+            } else if (NullChecker.isNullish(coreEnvelopeBatchSubmissionResponse.getTimeStamp())) {
+                LOG.error("CORE X12 Batch TimeStamp is empty...");
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Transforms the Batch Response message to Audit logging specific message.
+     *
+     * @param response
+     * @param assertion
+     * @param target
+     * @param direction
+     * @param _interface
+     * @param isRequesting
+     * @return LogEventRequestType
+     */
+    public static LogEventRequestType transformX12BatchMsgToAuditMsg(Object response, AssertionType assertion,
+        NhinTargetSystemType target, String direction, String _interface, boolean isRequesting) {
+        LOG.trace("Begin CoreX12AuditDataTransform -> transformX12BatchMsgToAuditMsg() -- NHIN");
+        LogEventRequestType result = null;
+        AuditMessageType auditMsg = null;
+        if (response == null) {
+            LOG.error("response Object was null");
+            return null;
+        }
+        if (assertion == null) {
+            LOG.error("Assertion was null");
+            return null;
+        }
+        // check to see that the required fields are not null
+        boolean missingReqFields = CoreX12AuditDataTransform.areRequiredCOREX12BatchFiledsNull(response, assertion);
+        if (missingReqFields) {
```

Comment:
Opening up a can of worms here, but I don't think this is right.  We fail to audit if any of these fields are missing?  I think we should be auditing what is there regardless.  So if user info is missing, for example, we fail to audit all the other information that is there?  I think we should discuss this.  Also that method that is used from XDR needs work.